### ISSUE TITLE
collocations

### ISSUE THREAD

**Issue by conjugateprior** [kbenoit/quanteda#767]
According to the help,`textstat_collocations` will 'Identify and score collocations from a corpus, character, or tokens object'. First argument description is `x: a character, corpus, or tokens object to be mined for collocations`.  But this function only works for tokens, and the character vectors shown that in the example. 

```r
res <- textstat_collocations(tokens(data_corpus_inaugural)) # works
textstat_collocations(data_corpus_inaugural)
Error in UseMethod("textstat_collocations") : 
  no applicable method for 'textstat_collocations' applied to an object of class "c('corpus', 'list')"
```

`collocations` "Detects collocations from texts or a corpus", except it doesn't work at all for me, even for tokens.

```{r}
collocations(data_corpus_inaugural)
[1] word1 word2 word3 count G2   
<0 rows> (or 0-length row.names)
collocations(tokens(data_corpus_inaugural))
[1] word1 word2 word3 count G2   
<0 rows> (or 0-length row.names)
```

Quanteda v.0.9.9-65 (CRAN)

**Comment by kbenoit** [comment]
Good points. We moved it to tokens only since we wanted the user, and not defaults for tokenisation, to make these decisions before scoring collocations. Could have been `tokens_collocations()` except it does not return a tokens object.

**Comment by conjugateprior** [comment]
Sounds reasonable, so for `textstat_collocations` it's just a matter of updating the help page.  

Is `collocations` supposed to work the same way?

**Comment by kbenoit** [comment]
`collocations()` is wrapped by `textstat_collocations()` and soon to be hidden. We're working on reimplementing the method more efficiently right now, should be done in mid-July if not sooner. 

So: Calling `collocations()` directly is deprecated.

**Comment by conjugateprior** [comment]
Right-o.  I see that currently it's 'sort of hidden': exported, and has a help page that doesn't mention it's deprecated, but that page doesn't appear in the index. (I ended up there by tab-complete). 

What's the official sequence for deprecating functions?

**Comment by kbenoit** [comment]
Normally to deprecate functions, we do the following:
* use `.Deprecated()` in the function call of the deprecated function, then pass it on to the new function. We remove the old function from the index, although it retains its man page until eliminated.
* for deprecated function arguments, we intercept the call, issue a manual warning, and call the function again with the correct arguments. In the CamelPocalyse of early 2017 we did this to the "remove*" arguments to `tokens()` for instance.

```r
> tokens("camel case arguments are deprecated", removePunct = TRUE)
tokens from 1 document.
Component 1 :
[1] "camel"      "case"       "arguments"  "are"        "deprecated"

Warning message:
In tokens.character("camel case arguments are deprecated", removePunct = TRUE) :
  removePunct is deprecated; use remove_punct instead
```

How we did this is in https://github.com/kbenoit/quanteda/blob/master/R/tokens.R#L189-L199.

Finally 
* all deprecated functions will be removed in 1.0 due by September. Before that we are likely to have a 0.9.9.9000 release with the old functions retained but all of the new API stable.
